### ISSUE TITLE
Use setuptools_scm for versioning 

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pluggy#73]
So we don't have to manually track versions.

Make sure to use method use by pytest of generating a `_version.py` file and importing it to obtain the version; the alternate method of using `pkg_resources` does not work from within frozen executables.

**Comment by tgoodlet** [comment]
A note on the [frozen executables problem](https://stackoverflow.com/a/14531668/4200924) as I wasn't familiar with it.

**Comment by nicoddemus** [comment]
I just realized that [PEP 562](https://docs.python.org/3.7/whatsnew/3.7.html#whatsnew37-pep562) will allow packages to make `__version__` lazy (if they want to provide one), which is great.

**Comment by nicoddemus** [comment]
Yep, and also with the nice side effect that you don't need to import `pkg_resources` and call the function to obtain the version of a package (I don't recall the name right now), which is quite slow.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus but with a version file you can just import it eagery to begin with, and currently we inconditionally import pkg_resources to begin with, so its no gain ^^

**Comment by RonnyPfannschmidt** [comment]
whops, i crosschecked, we dont

**Comment by nicoddemus** [comment]
Yep at the moment we use the `_version.py` trick

**Comment by tgoodlet** [comment]
@nicoddemus didn't this get done at some point?
